File: android/domain/src/main/kotlin/com/rideguard/domain/model/Platform.kt
package com.rideguard.domain.model

/**
 * A ride-hailing driver app we can read offers from.
 *
 * [packageName] is what goes in `accessibility_service_config.xml` so the
 * service stays completely dormant outside these apps — the single most
 * important battery decision in the whole app.
 */
enum class Platform(
    val packageName: String,
    val displayName: String,
    /**
     * Default platform take rate. Varies by country, driver tier and
     * promotion, so it is only a starting value — the driver overrides it in
     * settings after checking a real payout statement.
     */
    val defaultCommissionRate: Double,
    /**
     * Whether the fare shown on the offer card is ALREADY net of commission.
     * This differs by market and is the single easiest way to be quietly
     * wrong by 20%. Always verify against a real weekly statement.
     */
    val fareShownIsNetByDefault: Boolean,
) {
    /**
     * Romanian Bolt driver app spells it out on the card itself:
     *
     *     11,62 lei (NET, taxe incluse)
     *
     * Net, taxes already included. So there is nothing left to subtract. The
     * app previously defaulted this to gross with a 20% take rate, which made
     * every Bolt offer read 20% worse than it really was — a pessimistic HUD
     * is still a wrong HUD, and it would have talked the driver out of rides
     * that were actually fine.
     */
    BOLT(
        packageName = "ee.mtakso.driver",
        displayName = "Bolt",
        defaultCommissionRate = 0.0,
        fareShownIsNetByDefault = true,
    ),

    /**
     * Uber Romania labels the fare "Câștig net (fără comisionul Uber)" —
     * net earnings, Uber's commission already taken out. Same story.
     */
    UBER(
        packageName = "com.ubercab.driver",
        displayName = "Uber",
        defaultCommissionRate = 0.0,
        fareShownIsNetByDefault = true,
    ),
    UNKNOWN(
        packageName = "",
        displayName = "Unknown",
        defaultCommissionRate = 0.0,
        fareShownIsNetByDefault = true,
    ),
    ;

    companion object {
        fun fromPackage(pkg: String?): Platform =
            entries.firstOrNull { it.packageName.isNotEmpty() && it.packageName == pkg } ?: UNKNOWN

        /** Packages the accessibility service should listen to. */
        val watchedPackages: List<String> =
            entries.map { it.packageName }.filter { it.isNotEmpty() }
    }
}

File: ios/RideGuard/Core/Model/Platform.swift
import Foundation

/// A ride-hailing driver app we can read offers from.
///
/// `packageName` is meaningless as an iOS identifier — there is no way to ask
/// what app is in the foreground, let alone read it. It is kept because it is
/// the routing key `OfferParserRegistry` uses on both platforms, and because
/// fixtures recorded on Android must replay byte-identically here. On iOS the
/// value is synthesised from whichever platform the screenshot looks like.
///
/// Raw values are stable persistence keys: they end up in `UserDefaults` and
/// in the history JSON, so they must never be renamed even if the case names
/// change.
public enum Platform: String, CaseIterable, Codable, Sendable {
    case bolt = "BOLT"
    case uber = "UBER"
    case unknown = "UNKNOWN"

    public var packageName: String {
        switch self {
        case .bolt: return "ee.mtakso.driver"
        case .uber: return "com.ubercab.driver"
        case .unknown: return ""
        }
    }

    public var displayName: String {
        switch self {
        case .bolt: return "Bolt"
        case .uber: return "Uber"
        case .unknown: return "Unknown"
        }
    }

    /// Default platform take rate. Varies by country, driver tier and
    /// promotion, so it is only a starting value — the driver overrides it in
    /// settings after checking a real payout statement.
    public var defaultCommissionRate: Double {
        switch self {
        case .bolt: return 0.20
        case .uber: return 0.25
        case .unknown: return 0.0
        }
    }

    /// Whether the fare shown on the offer card is ALREADY net of commission.
    /// This differs by market and is the single easiest way to be quietly
    /// wrong by 20%. Always verify against a real weekly statement.
    public var fareShownIsNetByDefault: Bool {
        switch self {
        case .bolt: return false
        case .uber: return true
        case .unknown: return true
        }
    }

    public static func fromPackage(_ pkg: String?) -> Platform {
        allCases.first { !$0.packageName.isEmpty && $0.packageName == pkg } ?? .unknown
    }

    /// Packages a parser will accept. On Android this also feeds
    /// `accessibility_service_config.xml`; on iOS it is only the routing set.
    public static let watchedPackages: [String] =
        allCases.map(\.packageName).filter { !$0.isEmpty }

    /// The platforms a driver can actually pick in the UI — `unknown` is a
    /// parsing outcome, never a choice.
    public static let selectable: [Platform] = [.bolt, .uber]
}
